lib/ecto.ex

```diff
@@ -595,4 +595,25 @@ defmodule Ecto do
       true
     end
   end
+
+  def load(schema, data) when is_atom(schema) and is_map(data) do
+    map = stringify_map_keys(data)
+    Ecto.Schema.__load__(schema, nil, nil, nil, map, &Ecto.Type.load(&1, &2))
+  end
+
+  def load(schema, data) when is_list(data) do
+    load(schema, Map.new(data))
+  end
+
+  def load(types, data) when is_map(types) and is_map(data) do
+    map = stringify_map_keys(data)
+    Ecto.Schema.do_load(%{}, types, map, &Ecto.Type.load(&1, &2))
+  end
+
+  defp stringify_map_keys(map) do
+    Enum.into(map, %{}, fn
```

Comment:
Since we will end-up traversing the map inside `Ecto.Schema.__load__`, I would change `Ecto.Schema.__load__` to also accept a keyword list and use that in the default implementation. The map implementation can call `Map.to_list` and then delegate to the keyword one.